lib/rubocop/cop/style/redundant_condition.rb

```diff
@@ -196,6 +200,10 @@ def require_parentheses?(node)
             (node.respond_to?(:semantic_operator?) && node.semantic_operator?)
         end
 
+        def require_braces?(node)
+          node.hash_type?
```

Comment:
It will get a syntax error in the following already braced case:

```ruby
           if foo
           ^^^^^^ Use double pipes `||` instead.
             bar foo
           else
             bar({ baz => quux })
           end
```

Auto-corrected code is `bar foo || { { baz => quux } }`.

```ruby
% ruby -ce "bar foo || { { baz => quux } }"
-e:1: syntax error, unexpected '}', expecting =>
bar foo || { { baz => quux } }
```

Can you add the above test case and fix it?

Reply:
Thank you, fixed in 97c6273.